cmd/update_resource.go

```diff
@@ -21,100 +21,125 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"time"
-
 	"github.com/meroxa/cli/utils"
 	"github.com/meroxa/meroxa-go"
 	"github.com/spf13/cobra"
 )
 
+type UpdateResourceClient interface {
+	UpdateResource(ctx context.Context, key string, resourceToUpdate meroxa.UpdateResourceInput) (*meroxa.Resource, error)
+}
+
 type UpdateResource struct {
-	name, rType, metadata, credentials, url string
+	name, newName, metadata, credentials, url string
 }
 
-var updateResourceCmd UpdateResource
+func (ur *UpdateResource) setArgs(args []string) error {
+	if len(args) < 1 {
+		return errors.New("requires resource name")
+	}
+
+	ur.name = args[0]
+
+	return nil
+}
+
+func (ur *UpdateResource) setFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&ur.credentials, "credentials", "", "", "new resource credentials")
+	cmd.Flags().StringVarP(&ur.newName, "name", "", "", "new resource name")
+	cmd.Flags().StringVarP(&ur.metadata, "metadata", "m", "", "new resource metadata")
+	cmd.Flags().StringVarP(&ur.url, "url", "u", "", "new resource url")
+}
+
+func (ur *UpdateResource) execute(ctx context.Context, c UpdateResourceClient) (*meroxa.Resource, error) {
+	if ur.newName == "" && ur.url == "" && ur.metadata == "" && ur.credentials == "" {
+		return nil, errors.New("requires either `--credentials`, `--name`, `--metadata` or `--url` to update the resource")
+	}
+
+	if !flagRootOutputJSON {
+		fmt.Printf("Updating %s resource...\n", ur.name)
+	}
+
+	var res meroxa.UpdateResourceInput
+
+	// If name was provided, update it
+	if ur.newName != "" {
+		res.Name = ur.newName
+	}
+
+	// If url was provided, update it
+	if ur.url != "" {
+		res.URL = ur.url
+	}
+
+	// TODO: Figure out best way to handle creds and metadata
+	// Get credentials (expect a JSON string)
+	if ur.credentials != "" {
+		var creds meroxa.Credentials
+		err := json.Unmarshal([]byte(ur.credentials), &creds)
+		if err != nil {
+			return nil, err
```

Comment:
I think it would make sense to create a more user-friendly message here, something along the way of:
```suggestion
			return nil, fmt.Errorf("can't parse credentials: %w", err)
```
Same for the metadata parsed a few lines below.

Reply:
@lovromazgon good call